recommender/engines/anonymous/inference/anonymous_inference_component.py

```diff
@@ -0,0 +1,75 @@
+# pylint: disable=invalid-name, too-few-public-methods
+"""Recommender engine for anonymous users"""
+
+import random
+from typing import List, Tuple, Dict, Any
+
+from recommender.services.fts import retrieve_services_for_recommendation
+from recommender.engines.panel_id_to_services_number_mapping import PANEL_ID_TO_K
+from recommender.errors import (
+    InsufficientRecommendationSpaceError,
+    InvalidRecommendationPanelIDError,
+)
+
+
+class AnonymousInferenceComponent:
```

Comment:
I know this kind of change will require refactoring of the whole PR, but shouldn't this recommender be really called "Random" Inference component/pipeline?

Reply:
Like you specified in the PR description, this will also be called for not-anonymous users when we cannot find other engines

Reply:
Another thing: This should inherit the interface from BaseInferenceComponent. When you remove `self._for_logged_user` from BaseInferenceComponent and move the inference logic in `__call__`, you can now inherit from it here.

Reply:
Initially, I thought that there will be only 3 engines (NCF, RL, and Anonymous which will serve recommendations based on Google Analytics), but I agree that keeping a random engine has an advantage - it will be always available. So I will rename the engine to RandomInferenceComponent